### DISCUSSION TITLE
[HELP] Building to physical iOS Device

### DISCUSSION THREAD

**Discussion by BButner** [dotnet/maui#2364] [Q&A] [answered]
I've been looking at as many issues as I could find, the docs, etc.

Current Dev Environment:
- MacBook Pro (macOS Big Sur 11.5.2)
- iPhone XS Max (iOS Beta 15 Release 7)

I've gotten my simple MAUI app working just fine, and building/running to the iOS Simulator for iPad/iPhone, but now I would like to build it to my physical device. I've gotten the device provisioned and added to the devices list under the Apple Developer portal, but I'm having an issue figuring out the proper command to actually build it and run it to my physical iPhone.

I found [this](https://github.com/dotnet/xamarin/issues/26#issuecomment-849735553) post by @rolfbjarne, but unfortunately that does not seem to work in my case (perhaps the syntax is different?).

The exact command I'm running is:

```
dotnet build [PROJECTNAME] -t:Run -f net6.0-ios -p:_DeviceName=[UDID]
```

I'm new to macOS/iOS dev, so if I left any important information out please let me know and I'll provide it!

**Comment by rolfbjarne** [comment, marked as answer]
You need to set the RuntimeIdentifier property to a device architecture. This is typically `ios-arm64` for iOS devices.

This can be done entirely from the command line like this:

> dotnet build [PROJECTNAME] -t:Run -f net6.0-ios -p:_DeviceName=[UDID] /p:RuntimeIdentifier=ios-arm64

alternatively you can add this in the first PropertyGroup in your project file:

```xml
<RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
```

the disadvantage of the last solution is that now you'll have the opposite problem for the simulator (by default we set `RuntimeIdentifier=iossimulator-x64`, which means you'll have to set it manually on the command line if you've set it to a device architecture in the project file).

**Reply by BButner** [reply]
Rolf,

Thanks for the reply, that worked like a charm! I ran into another issue where it was claiming `A valid provisioning profile for this executable was not found`, however that turned out to be caused by the fact that I had an iOS 15 Beta Provisioning Profile from Apple installed on my iPhone XS Max. I grabbed an iPad (that has had no previous provisioning profile installed) and it worked without issue!

Did I miss this `RuntimeIdentifier` `iOS-arm64` vs `iossimulator-x64` somewhere in some sort of documentation?

Thanks again!

**Reply by rolfbjarne** [reply]
> Did I miss this `RuntimeIdentifier` `iOS-arm64` vs `iossimulator-x64` somewhere in some sort of documentation?

No, not really, we haven't gotten around to documenting everything yet (also if you're using Visual Studio, this will be handled automatically for you).
